tests/integration/test_nvt_tf_training.py

```diff
@@ -0,0 +1,452 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import glob
+import math
+
+# External dependencies
+import os
+import shutil
+from os import path
+from os.path import dirname, realpath
+
+import cudf
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from sklearn.model_selection import train_test_split
+from test_notebooks import _run_notebook
+
+import nvtabular as nvt
+from nvtabular import ops
+from nvtabular.inference.triton import export_tensorflow_ensemble
+from nvtabular.utils import download_file
+
+TEST_N_ROWS = 64
+
+
+def test_nvt_tf_movielens_training():
```

Comment:
Is this also any different from the movielens notebooks?

Reply:
Similarly, this test has the extra code that runs NVT transform and TF predict.

Reply:
How about we put the nvt transform/ tf predict code to run in the other test script? That way we can eliminate most of the boilerplate/duplicate code here and just run the notebook.  The workflow and tensorflow model are both saved so this should be pretty easy

Reply:
@benfred  I was planning to do something similar when I first started to this but one of my concern on this approach was that sometimes the notebook based integration tests fail due to some reasons (i.e. file name change, etc). And, this is one of the reasons I had this duplicate code. With this way, I know that it only fails if there is any change in the APIs. Also, I can make any changes easily if I have to (like adding extra randomly generated columns for testing a particular feature. We did this for the HugeCTR integration test using movielens example). 

I still think we should keep it as it is even though most of the them is duplicate code right now. I really like these integration tests are independent from the notebooks for easy changeability whenever we need it for testing purposes.

Also, we might wanna switch the original movielens and rossmann data with randomly generated data in the future as well to test certain features whenever we want to.

What do you think?

Reply:
@benfred I am trying to put only the nvt transform/ tf predict code in a separate test but I am hitting an error when I try to load the trained tf model to run predict. When I googled the error, I found out that other developers also hit the same error with TF version 2.3.

Here is a link about the error;

https://github.com/tensorflow/tensorflow/issues/43478

What do you suggest we should do?